robot/impl/resource_manager_modular_test.go

```diff
@@ -447,5 +453,8 @@ func (m *dummyModMan) ValidateConfig(ctx context.Context, cfg resource.Config) (
 }
 
 func (m *dummyModMan) Close(ctx context.Context) error {
+	if len(m.state) != 0 {
```

Comment:
did the tests fail with this when you "undid" your fix?

Reply:
Yes, they did fail. It was convenient that this was already _almost_ set up to catch the issue.